plugins/inputs/influxdb_listener/influxdb_listener.go

```diff
@@ -180,26 +182,24 @@ func (h *InfluxDBListener) Start(acc telegraf.Accumulator) error {
 	h.listener = listener
 	h.port = listener.Addr().(*net.TCPAddr).Port
 
-	h.wg.Add(1)
 	go func() {
-		defer h.wg.Done()
-		h.server.Serve(h.listener)
+		err = h.server.Serve(h.listener)
+		if err != http.ErrServerClosed {
+			h.Log.Infof("Error serving HTTP on %s", h.ServiceAddress)
+		}
 	}()
 
-	h.log.Infof("Started HTTP listener service on %s", h.ServiceAddress)
+	h.Log.Infof("Started HTTP listener service on %s", h.ServiceAddress)
 
 	return nil
 }
 
 // Stop cleans up all resources
 func (h *InfluxDBListener) Stop() {
-	deadline := time.Now().Add(10 * time.Second)
-	ctx, cancel := context.WithDeadline(context.Background(), deadline)
-	defer cancel() // go vet requires us to call cancel even though we're waiting for the deadline
-	h.server.Shutdown(ctx)
-	h.wg.Wait()
-
-	h.log.Infof("Stopped HTTP listener service on %s", h.ServiceAddress)
+	err := h.server.Shutdown(context.Background())
```

Comment:
I'm a bit concerned about having no deadline at all. If something doesn't close cleanly, it means the service could just hang on close?

Reply:
I think the agent could give up on waiting for the plugin after a period of time instead of handling that in plugin.  We could consider adding this in a different PR.

The ReadTimeout option should be the maximum amount of time a request can run, and I think that is the max time we can spend here.

Reply:
When we call Shutdown it closes its internal context and I modified handleWrite to check that context before parsing each metric.  With that check I don't know of anything that would cause graceful shutdown to block.